### ISSUE TITLE
CV:  Fight bugs and enhance generic dataset template

### ISSUE THREAD

**Issue by tilfischer** [ComPlat/chemotion_ELN#1750]
- I am an ELN syetem admin and user

# Description
While adding metadata for CV measurements via the corresponding generic dataset with ID 7cff325e-b628-4607-8d36-a11ddd7abdb1 fetched from the [LabIMotion Template Hub](https://www.chemotion-repository.net/home/genericHub) I noticed a bug for "Purge electrolyte solution". When checking the checkbox, the three fields for the type of gas, flow rate and time are not shown. Moreover, for further fields (Supporting electrolyte, Atmosphere) which have a controlled list where users my select "others" the corresponding additional field to specify "other something" is not shown. 

I have made the changes to the generic dataset to fix the bugs. Moreover, there were quite a number of further aspects I tried to fix (see list below). I released a minor version. The JSON v0.2 is attached here:

[DatasetKlass_cyclic voltammetry (CV)_d1eaea70-e3f4-4f85-aabc-51c3fa7d5eaf.json](https://github.com/ComPlat/chemotion_ELN/files/14240210/DatasetKlass_cyclic.voltammetry.CV._d1eaea70-e3f4-4f85-aabc-51c3fa7d5eaf.json)

I provide a list of things I changed, so that we could also discuss this here.

# Bugs
- The bug with  got fixed (see JSON). If the checkbox "Purge electrolyte solution" is checked, the additional fields for type of gas, flow rate and time were not shown. The three fields on the type of gas, flow rate and time should be displayed now, if the checkbox is checked (see JSON and video).
- For electrolyte there is a controlled list with the option "others". If this is selected the field salt_other should be possibly shown, as this is part of the generic dataset template, but it was not. Fixed.

# Enhancements
- DisplayName:data_type -> DisplayName:Type of data
- DisplayName:operator -> DisplayName:Operator
- DisplayName:Creator -> DisplayName:Author/Creator
- The fields for Author/Creator and Contributor are automatically filled. I guess there is also data on ORCID iDs available to also add this information. Hence, I added field for Author/Creator ORCID iD and Contributor ORCID iD with placeholder 0000-0000-0000-0000-0000. Consequently, I also added a field for Operator ORCID iD with placeholder 0000-0000-0000-0000-0000.
- DisplayName:label -> DisplayName:Label
- DisplayName:id -> DisplayName:ID
- While Volume was type system-defined with option_layers:volume, the amount was not. I changed for amount from text to system-defined with option_layer:mass. 
- Is the field "size" under "Sample preparation" needed? The size of the sample should not matter anymore, if it gets dissolved. I was that brave to remove this field. 
- As there is a controlled list for supporting electrolytes, I added a controlled list for solvents also allowing "others", which opens another field to specify the solvent.
- DisplayName:Concentration -> DisplayName:Concentration of analyte
- DisplayName:Amount electrolyte -> DisplayName:Amount supporting electrolyte
- DisplayName:Concentration electrolyte -> DisplayName:Concentration supporting electrolyte
- In the controlled list of supporting electrolytes there is TBA[PF6] and TBABF4. I made adoptions to that list and removed the square brackets to have it uniform.
- DisplayName:instrument -> DisplayName:Instrument
- For used electrodes I moved the traditional disk electrode to the top of the controlled list.
- What should be entered for ECSA method was totally unclear to use. The area can be determined by a measurement and reported, if this is done. From another resource I found out that people added "geometric" to this field, as the ECSA was simply calculated from the diameter of the electrode. Consequently, I added geometric as placeholder. The previous placeholder was also to long and was not completely displayed. 
- Also in this other resource, someone added "8 figure" to "Polishing method". I added "8 figure-eight" as placeholder to that field. 
- Added diamond as polishing material controlled list.
- Within the section on "Additional equipment and devices", there is the field "reference" with FieldName:Reference electrode (RE), which has a controlled list of reference electrodes. If "other" is selected the field "other_re" is shown. This can be possibly used for exotic examples. One reference which is not exotic should be, however, added to the controlled list, which is "Ag (QRE)" as another option see https://doi.org/10.1021/ic50211a080. Note: QRE is an abbreviation used in the [IUPAC Recommendations of electrochemical methods and analysis](https://doi.org/10.1515/pac-2018-0109) from 2019. The old term "pseudo-reference electrode" is deprecated. 
- Within the section on "Setting describing the measurement", there is a field named "Conditions" and one may select "illuminated" and "darkness". Another option which is missing is simply "daylight". I added this to the controlled list. 
- Within the section on "Additional equipment and devices", there is the field "reference" with FieldName:Reference electrode (RE), which has a controlled list of reference electrodes. One option is "Ag/AgNO3 in MeCN" and AgNO3 is very soluble in MeCN i.e. not comparable to "Ag/AgCl in H2O", where AgCl is almost insoluble. As the potential of this Ag/AgNO3 in MeCN electrode considerably depends on the concentration of AgNO3, see this reference https://doi.org/10.1016/S0020-1693(99)00407-7, more options need to be added. Similar enhancements could be done for Ag/AgCl electrodes in the list where, compared to Ag/AgNO3, no solvent is specified. I added "H2O" to the entries, and added the electrodes from the reference linked above (see JSON) and this list:
   - Ag/AgCl (KCl (sat.) in H2O)
   - Ag/AgCl (3 M KCl in H2O)
   - Ag/AgCl (NaCl (sat.) in H2O)
   - Ag/AgCl (3 M NaCl in H2O)
   - Ag/AgNO3 (0.1 M AgNO3, 0.1 M TBAPF6 in MeCN)
   - Ag/AgNO3 (0.01 M AgNO3, 0.1 M TBAPF6 in MeCN)
   - Ag/AgNO3 (0.001 M AgNO3, 0.1 M TBAPF6 in MeCN)
   - Ag/AgClO4 (0.01 M AgClO4, 0.1 M TBAPF6) in MeCN
   - SCE (KCl (sat.) in H2O)
   - SCE (3 M KCl in H2O)
   - SSCE (NaCl (sat.) in H2O)
   - SHE
   - NHE
   - RHE
   - Ag (QRE)
   - others
- DisplayName:IR correction -> DisplayName:iR correction
- There were two field with the name "reference". For internal reference, I changed this to internal_reference with internal_reference_others. I also allowed to select none.
- Atmosphere now also allows to select "Air", which was added to atm list.

Here is a video preview of this suggested generic dataset template for CV v0.2: 

Uploading chrome_7Ic2H4n506.mp4…

Best,
Tillmann

PS: Particle size has option_layers: lenght, which as units m, cm, mm. Also µm should be added. Possibly this is another issue...

**Comment by herrdivad** [comment]
Hi @tilfischer,
a new version 0.3 should now be online on the hub. Could you test it for your Instance please (I will test the version on Complat ELN and Repo in parallel).
We tried to migrate everything but not touched the blue category, because they should be generic across all Layouts and some of them are also automatically (backended). According to @nicolejung we should and could please go into more discussion if you like to have some changes in the long run.
